query.go

```diff
@@ -917,6 +918,9 @@ func NewQueryProperty(key string, value *string, captureId *uint) QueryProperty
 
 // Next will return the next match in the sequence of matches.
 //
+// Subsequent calls to [QueryMatches.Next] will reuse the same memory,
+// this means that you need to copy/take out what you need from the [QueryMatch] before calling [QueryMatches.Next] again.
+//
```

Comment:
```suggestion
// Subsequent calls to [QueryMatches.Next] will overwrite the memory at the same location as prior matches, since the memory is reused.
// If you need to keep the data of a prior match without it being overwritten, you should copy what you need before calling [QueryMatches.Next] again.
//
```